src/data-card-advanced.tsx

```diff
@@ -0,0 +1,463 @@
+import {vars} from './skins/skin-contract.css';
+import IconCloseRegular from './generated/mistica-icons/icon-close-regular';
+import IconButton from './icon-button';
+import Inline from './inline';
+import classNames from 'classnames';
+
+import type StackingGroup from './stacking-group';
+import type Image from './image';
+// import type {
+//     ValueBlock,
+//     InformationBlock,
+//     ProgressBlock,
+//     HighlightedValueBlock,
+//     RowBlock,
+//     SimpleBlock,
+// } from './advanced-slots';
+import type {ButtonPrimary, ButtonLink} from './button';
+import type {DataAttributes, IconProps} from './utils/types';
+import type {RendersNullableElement} from './utils/renders-element';
+
+type CardAction = {
+    label: string;
+    onPress: () => void;
+    Icon?: React.FC<IconProps>;
+    iconSize?: number;
+    iconColor?: string;
+    iconBackground?: string;
+    iconBackgroundInverse?: string;
+};
+
+type CardActionsGroupProps = {
+    actions: Array<CardAction>;
+    isInverse?: boolean;
+};
+
+const CardActionsGroup = ({actions, isInverse}: CardActionsGroupProps): JSX.Element => {
+    return (
+        <Inline space={0}>
+            {actions.map(
+                (
+                    {
+                        onPress,
+                        label,
+                        Icon,
+                        iconSize = 20,
+                        iconColor = vars.colors.neutralHigh,
+                        iconBackground = styles.cardAction,
+                        iconBackgroundInverse = styles.cardActionInverse,
+                    },
+                    index
+                ) =>
+                    Icon ? (
+                        <IconButton
+                            size={48}
+                            key={index}
+                            onPress={onPress}
+                            aria-label={label}
+                            className={styles.cardActionIconButton}
+                            style={{display: 'flex'}}
+                        >
+                            <div className={isInverse ? iconBackgroundInverse : iconBackground}>
+                                <Icon color={iconColor} size={iconSize} />
+                            </div>
+                        </IconButton>
+                    ) : (
+                        <div key={index} className={styles.cardActionIconButton} />
+                    )
+            )}
+        </Inline>
+    );
+};
+
+const useTopActions = (actions?: Array<CardAction>, onClose?: () => void) => {
+    const {texts} = useTheme();
+    const finalActions = actions ? [...actions] : [];
+
+    if (onClose) {
+        finalActions.push({
+            label: texts.closeButtonLabel,
+            onPress: onClose,
+            Icon: IconCloseRegular,
+        });
+    }
+
+    return finalActions;
+};
+
+type CardContentProps = {
+    headline?: string | RendersNullableElement<typeof Tag>;
+    pretitle?: string;
+    pretitleAs?: string;
```

Comment:
what's this used for?

Reply:
It was requested that there be the possibility of changing the pretitle and title, to h1 to h4, while keeping the default values ​​like p to pretitle and h3 to title, that's why the purpose of these props As.